Source/NSTreeController.m

```diff
@@ -127,123 +155,242 @@ - (BOOL) selectsInsertedObjects
   return _selectsInsertedObjects;
 }
 
-- (BOOL) setSelectionIndexPath: (NSIndexPath*)indexPath
+- (BOOL) setSelectionIndexPath: (NSIndexPath *)indexPath
 {
-  // FIXME
-  return NO;
+  BOOL f = [self commitEditing];
+
+  if (YES == f)
+    {
+      [_selection_index_paths addObject: indexPath];
```

Comment:
The name of the method rather suggests that it replaces the old content. Just like the next method.

Reply:
Ah, yes, okay... fixing.